keras/models.py

```diff
@@ -76,6 +88,49 @@ def standardize_weights(y, sample_weight=None, class_weight=None):
     else:
         return np.ones(y.shape[:-1] + (1,))
 
+def sequential_from_yaml(yamlString):
+    '''
+        Returns a compiled Sequential model generated from a local yaml file,
+        which is either created by hand or from Sequential.to_yaml
+    '''
+    modelYaml = yaml.load(yamlString)
+    model = Sequential()
+
+    class_mode = modelYaml.get('class_mode')
+    theano_mode = modelYaml.get('theano_mode')
+    loss = globals()[modelYaml.get('loss')]
+
+    optim = modelYaml.get('optimizer')
+    optimName = optim.get('name')
+    optim.pop('name')
+    optimizer = globals()[optimName](**optim)
+
+    layers = modelYaml.get('layers')
+    for layer in layers:
+        name = layer.get('name')
+        layer.pop('name')
+        hasParams = False
+        if layer.has_key('parameters'):
+            params = layer.get('parameters')
+            layer.pop('parameters')
+            hasParams = True
+        for k, v in layer.iteritems():
+            if isinstance(v, dict):
+                vname = v.get('name')
+                v.pop('name')
+                layer[k] = globals()[vname](**v)
```

Comment:
This seems dangerous. I would definitely prefer a more verbose, more explicit approach. First determine if the object is a constraint/regularizer/etc, then get it from the appropriate module.